src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsFrameworkParser.cs

```diff
@@ -25,6 +25,7 @@ public interface IVsFrameworkParser
         /// <exception cref="ArgumentNullException">If the provided string is null.</exception>
         /// <exception cref="ArgumentException">If the provided string cannot be parsed.</exception>
         /// <returns>The parsed framework.</returns>
+        [Obsolete("This API does not support .NET 5 and higher target frameworks with platforms. Use IVsFrameworkParser2 instead.")]
```

Comment:
Why not deprecate the whole `IVsFrameworkParser` interface? All methods (2) of this interface will be deprecated in this PR.